database/src/main/postgres/flows/V1.7.5__get_flow_checkpoints.sql

```diff
@@ -0,0 +1,127 @@
+    OUT status                INTEGER,
+    OUT status_text           TEXT,
+    OUT id_checkpoint         UUID,
+    OUT checkpoint_name       TEXT,
+    OUT measure_name          TEXT,
+    OUT measure_columns       TEXT[],
+    OUT measurement_value     JSONB,
+    OUT checkpoint_start_time TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_end_time   TIMESTAMP WITH TIME ZONE
+) RETURNS SETOF record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: flows.get_flow_checkpoints(3)
+--      Retrieves all checkpoints (measures and their measurement details) related to a primary flow
+--      associated with the input partitioning.
+--
+-- Note: a single row returned from this function doesn't contain all data related to a single checkpoint - it only
+--     represents one measure associated with a checkpoint. So even if only a single checkpoint would be retrieved,
+--     this function can potentially return multiple rows.
+--
+-- Note: checkpoints will be retrieved in ordered fashion, by checkpoint_time and id_checkpoint
+--
+-- Parameters:
+--      i_partitioning_of_flow  - partitioning to use for identifying the flow associate with checkpoints
+--                                that will be retrieved
+--      i_limit                 - (optional) maximum number of checkpoints retrieved
+--                                if zero or lower specified, all data will be returned, i.e. no limit will be applied
+--      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
+--
+--      Note: checkpoint name uniqueness is not enforced by the data model, so there can be multiple different
+--          checkpoints with the same name in the DB, i.e. multiple checkpoints can be retrieved even when
+--          specifying `i_checkpoint_name` parameter
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_checkpoint       - id of retrieved checkpoint
+--      checkpoint_name     - name of retrieved checkpoint
+--      measure_name        - measure name associated with a given checkpoint
+--      measure_columns     - measure columns associated with a given checkpoint
+--      measurement_value   - measurement details associated with a given checkpoint
+--      checkpoint_time     - time
+--
+-- Status codes:
+--      11                  - Flow created
+--      16                  - No checkpoints found
+--      41                  - Partitioning not found
+--
+--
+-------------------------------------------------------------------------------
+
+DECLARE
+    _fk_partitioning BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning_of_flow);
+
+    IF _fk_partitioning IS NULL THEN
+        status := 41;
+        status_text := 'The partitioning does not exist.';
+        RETURN NEXT;
+        RETURN;
+    END IF;
+
+    RETURN QUERY
+    SELECT 11 AS status, 'OK' AS status_text,
```

Comment:
Matter of taste, but this alignment is... 🙅 😄

Reply:
what would be better? :D

Reply:
Didn't we talk about it? 😉 
```sql
SELECT *
FROM my_table
    INNER JOIN your_table
WHERE true
```
or if you prefer
```sql
SELECT *
FROM   my table
    INNER JOIN your_table
WHERE  true
```

For the record, I don't want to force you to change it. Just mentioning this one is... hurting me 🤣

Reply:
We talked and the outcome I gathered out of it was that we want left align style - in both ways, that `INNER JOIN` line is in my view a bit nasty, but indeed this is a personal style. (I'm used to https://www.sqlstyle.guide/ but left alignment is not a hill I want to die on :D )

Reply:
actually :( it doesn't look right (to me) when I change it like this: (a partial snippet)

```sql
    SELECT 11 AS status, 'OK' AS status_text,
           CP.id_checkpoint, CP.checkpoint_name,
           MD.measure_name, MD.measured_columns,
           M.measurement_value,
           CP.process_start_time AS checkpoint_start_time, CP.process_end_time AS checkpoint_end_time
    FROM   runs.checkpoints AS CP
    JOIN   runs.measurements AS M
      ON   CP.id_checkpoint = M.fk_checkpoint
    JOIN   runs.measure_definitions AS MD
      ON   M.fk_measure_definition = MD.id_measure_definition
    WHERE  EXISTS (
             -- Identification of a flow and all partitionings related to it.
             SELECT 1
             FROM   flows.flows AS F
             JOIN   flows.partitioning_to_flow AS PF
               ON   F.id_flow = PF.fk_flow
             WHERE  F.fk_primary_partitioning = _fk_partitioning
               AND  CP.fk_partitioning = PF.fk_partitioning
           )
      AND  (i_checkpoint_name IS NULL OR CP.checkpoint_name = i_checkpoint_name)
```